### PR TITLE
Removed conflicts from Jake Archibald pull request

### PR DESCRIPTION
**Pull request by ianbjacobs**
https://github.com/w3c/payment-handler/pull/305

The following tasks have been completed:

 * [ ] web platform tests (link)
 * [ ] MDN Docs added (link)

Implementation commitment:

 * [ ] Safari (link to issue)
 * [ ] Chrome (link to issue)
 * [ ] Firefox (link to issue)
 * [ ] Edge (public signal)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-handler/pull/327.html" title="Last updated on Oct 17, 2018, 6:22 PM GMT (0cc8937)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-handler/327/e63b2d3...0cc8937.html" title="Last updated on Oct 17, 2018, 6:22 PM GMT (0cc8937)">Diff</a>

### CODE REVIEW

**Comment by ianbjacobs** [comment]
Hi @rsolomakhin and @jakearchibald,

I endeavored to resolve the merge conflicts between Jake's pull request:
 https://github.com/w3c/payment-handler/pull/305

and the current draft of the spec. 

@rsolomakhin, I removed the bits you suggested be removed. I did not (yet) review the full algorithm to see whether it still makes sense in light of the edits.

Ian

File: index.html [hunk 1]
```diff
@@ -1555,56 +1562,45 @@ <h2>
...
+                serialization of the origin</a> of the context where
+                PaymentRequest was initialized.
+              </dd>
+              <dt>
+                <a data-lt="PaymentRequestEvent.methodData">methodData</a>
+              </dt>
+              <dd>
+                The result of executing the <a>MethodData Population
+                Algorithm</a>
+              </dd>
```
Comment (rsolomakhin) [inline index.html:1633]: Let's keep `total`, `paymentRequestId`, `instrumentKey`, and `modifiers` in the `PaymentRequestEvent` for now.

```suggestion
              </dd>
              <dt>
                <a data-lt="PaymentRequestEvent.total">total</a>
              </dt>
              <dd>
                A <a>structured clone</a> of the total field on
                the <a>PaymentDetailsInit</a> from the corresponding
                <a>PaymentRequest</a>.
              </dd>
              <dt>
                <a data-lt="PaymentRequestEvent.paymentRequestId">paymentRequestId</a>
              </dt>
              <dd>
                The [[\details]].<a>id</a> from the <a>PaymentRequest</a>.
              </dd>
              <dt>
                <a data-lt="PaymentRequestEvent.instrumentKey">instrumentKey</a>
              </dt>
              <dd>
                The <a>instrumentKey</a> of the selected <a>PaymentInstrument</a>,
                or the empty string if none was selected.
              </dd>
              <dt>
                <a data-lt="PaymentRequestEvent.modifiers">modifiers</a>
              </dt>
              <dd>
                The result of executing the <a>Modifiers Population Algorithm</a>.
              </dd>
```

**Review by rsolomakhin — COMMENTED** [review]
Looks good, but need to preserve the fields in `PaymentRequestEvent`.

File: index.html [hunk 2]
```diff
@@ -935,31 +935,69 @@ <h2>
...
+                <a>PaymentInstrument</a>, or the empty string if none was
+                selected.
+              </dd>
+              <dt>
+                <a data-lt="PaymentRequestEvent.modifiers">modifiers</a>
+              </dt>
+              <dd>
+                The result of executing the <a>Modifiers Population
+                Algorithm</a>.
+              </dd>
```
Comment (rsolomakhin) [inline index.html:975]: Could you move these fields from `canmakepayment` event to `paymentrequest` event, please?
 - total
 - paymentRequestId
 - instrumentKey

File: index.html [hunk 3]
```diff
@@ -935,31 +935,69 @@ <h2>
...
-            </ol>
+          <li>
+            <p>
+              <a>Fire Functional Event</a> "<code>canmakepayment</code>" using
+              <a>CanMakePaymentEvent</a> on <var>registration</var> with the
+              following properties:
+            </p>
+            <dl>
+              <dt>
+                <a data-lt="PaymentRequestEvent.topOrigin">topOrigin</a>
```
Comment (rsolomakhin) [inline index.html]:
```suggestion
                <a data-lt="CanMakePaymentEvent.topOrigin">topOrigin</a>
```

File: index.html [hunk 4]
```diff
@@ -935,31 +935,69 @@ <h2>
...
+              </dt>
+              <dd>
+                <a data-cite="!HTML#ascii-serialisation-of-an-origin">the
+                serialization of the origin</a> of the top level payee web
+                page.
+              </dd>
+              <dt>
+                <a data-lt=
+                "PaymentRequestEvent.paymentRequestOrigin">paymentRequestOrigin</a>
```
Comment (rsolomakhin) [inline index.html]:
```suggestion
                "CanMakePaymentEvent.paymentRequestOrigin">paymentRequestOrigin</a>
```

File: index.html [hunk 5]
```diff
@@ -935,31 +935,69 @@ <h2>
...
+              </dt>
+              <dd>
+                <a data-cite="!HTML#ascii-serialisation-of-an-origin">the
+                serialization of the origin</a> of the context where
+                PaymentRequest was initialized.
+              </dd>
+              <dt>
+                <a data-lt="PaymentRequestEvent.methodData">methodData</a>
```
Comment (rsolomakhin) [inline index.html]:
```suggestion
                <a data-lt="CanMakePaymentEvent.methodData">methodData</a>
```

File: index.html [hunk 6]
```diff
@@ -935,31 +935,69 @@ <h2>
...
+                <a data-lt=
+                "PaymentRequestEvent.instrumentKey">instrumentKey</a>
+              </dt>
+              <dd>
+                The <a>instrumentKey</a> of the selected
+                <a>PaymentInstrument</a>, or the empty string if none was
+                selected.
+              </dd>
+              <dt>
+                <a data-lt="PaymentRequestEvent.modifiers">modifiers</a>
```
Comment (rsolomakhin) [inline index.html]:
```suggestion
                <a data-lt="CanMakePaymentEvent.modifiers">modifiers</a>
```

**Review by rsolomakhin — COMMENTED** [review]
Almost missed this :-)

The field names in `canmakepayment` event should be associated with `CanMakePaymentEvent`, not the `PaymentRequestEvent`.

**Review by rsolomakhin — APPROVED** [review]
LGTM!

**Pull request merged**